LogicForPad.java

```diff
@@ -0,0 +1,159 @@
+            pjob.print();
+        } catch (PrinterException pe) {
+            JOptionPane.showMessageDialog(null,
+                    "Printer error" + pe, "Printing error",
+                    JOptionPane.ERROR_MESSAGE);
+        }
+    }
+	
+    //Method that allows for simple replace, where you select code to replace and type in what you want 
+    //to replace it
+    public void simpleReplace(JTextPane fileContent){
+    	String input=JOptionPane.showInputDialog("Replace or insert with:");
+    	if(input!=null)
+    		fileContent.replaceSelection(input);
+    }
+    
+  //Takes in a file and opens its contents in the notepad
+    public void openFile(File fileToOpen, JTextPane fileContent) {
+    	StringBuilder content=new StringBuilder();
+    	try {
+            BufferedReader in = new BufferedReader(new FileReader(fileToOpen));
+            String line;
+            while ((line = in.readLine()) != null){
+            	content.append(line+"\n");
+            }
+            fileContent.setText(content.toString());
+            JOptionPane.showMessageDialog(null, "File is opened successfully...");
+            in.close();
+            addToRecent(fileToOpen);
+        } catch (IOException ex) {
+        }
+    }
+  //Takes in a file and adds it to the recently opened files array. Drops the oldest file 
+    //if array has 5 or more items in it.
+    public void addToRecent(File file) {
+    	if(!recentFiles.contains(file)) {
+    		if(recentFiles.size()>=5) {
+    			recentFiles.remove(0);
+    		}
+    		recentFiles.add(file);
+    	}
+    }
+  //Takes in an integer which represents which file in the recently opened array to open, and opens that file
+    public void openRecent(int x,JTextPane fileContent) {
+    	File fileToOpen=recentFiles.get(x);
+    	openFile(fileToOpen,fileContent);
+    }
+    
+  //Opens a file in the note pad
+    public void openAndSelectFile(JTextPane fileContent) {
+    	File fileToOpen = null;
+        JFileChooser fc = new JFileChooser();
+        int returnVal = fc.showOpenDialog(null);
+        if (returnVal == JFileChooser.APPROVE_OPTION)
+            fileToOpen = fc.getSelectedFile();
+        openFile(fileToOpen,fileContent);
+    }
+  //Takes in two strings and returns a menu item using the first string 
+    //as the text and the second as the action command
+    public JMenuItem makeMenuItem(String name, ActionListener al) {
+    	JMenuItem menuItem = new JMenuItem(name);
+        menuItem.addActionListener(al);
+        return menuItem;
+    }
+    
+  //This method updates the menu displaying recently opened files
+    public void updateRecent(JTextPane fileContent,JMenu openRecentSelect) {
+    	openRecentSelect.removeAll();
+    	int x=0;
+    	for(File file:recentFiles) {
+    		final int ItemIndex=x;
+        	JMenuItem recents=makeMenuItem(file.getName(),(e)->{openRecent(ItemIndex,fileContent);});
+        	x++;
+        	openRecentSelect.add(recents);
+        }
+    }

```

Comment:
It seems to me that the functions for the File menu and the functions for the Edit menu are not separated clearly in this file. You should add a comment specifying the sections